perun-core/src/test/java/cz/metacentrum/perun/core/entry/ServicesManagerEntryIntegrationTest.java

```diff
@@ -1295,709 +1292,6 @@ public void getDestinationsWhenFacilityNotExists() throws Exception {
-		perun.getResourcesManager().assignGroupToResource(sess, group, resource, false, false, false);
-		perun.getResourcesManager().assignGroupToResource(sess, group2, resource, false, false, false);
-
-		perun.getGroupsManager().setMemberGroupStatus(sess, member2, group, MemberGroupStatus.EXPIRED);
-		perun.getGroupsManager().setMemberGroupStatus(sess, member3, group, MemberGroupStatus.EXPIRED);
-		perun.getGroupsManager().setMemberGroupStatus(sess, member3, group2, MemberGroupStatus.EXPIRED);
-
-		// set element's name/id as required attributes to get some attributes for every element
-		Attribute reqFacAttr;
-		reqFacAttr = perun.getAttributesManager().getAttribute(sess, facility, "urn:perun:facility:attribute-def:core:name");
-		perun.getServicesManager().addRequiredAttribute(sess, service, reqFacAttr);
-		Attribute reqResAttr;
-		reqResAttr = perun.getAttributesManager().getAttribute(sess, resource, "urn:perun:resource:attribute-def:core:name");
-		perun.getServicesManager().addRequiredAttribute(sess, service, reqResAttr);
-		Attribute reqGrpAttr;
-		reqGrpAttr = perun.getAttributesManager().getAttribute(sess, group, "urn:perun:group:attribute-def:core:name");
-		perun.getServicesManager().addRequiredAttribute(sess, service, reqGrpAttr);
-		Attribute reqMemAttr;
-		reqMemAttr = perun.getAttributesManager().getAttribute(sess, member, "urn:perun:member:attribute-def:core:id");
-		perun.getServicesManager().addRequiredAttribute(sess, service, reqMemAttr);
-		Attribute reqVoAttr;
-		reqVoAttr = perun.getAttributesManager().getAttribute(sess, vo, "urn:perun:vo:attribute-def:core:name");
-		perun.getServicesManager().addRequiredAttribute(sess, service, reqVoAttr);
-
-		// finally assign service
-		service.setUseExpiredMembers(false);
-		perun.getServicesManagerBl().updateService(sess, service);
-		perun.getResourcesManager().assignService(sess, resource, service);
-
-		List<ServiceAttributes> facilities = new ArrayList<ServiceAttributes>();
-		facilities.add(perun.getServicesManager().getDataWithVos(sess, service, facility));
-		List<ServiceAttributes> vos = facilities.get(0).getChildElements();
-		List<ServiceAttributes> resources = vos.get(0).getChildElements();
-		List<ServiceAttributes> resourceElements = resources.get(0).getChildElements();
-		List<ServiceAttributes> groupMembers = resourceElements.get(0).getChildElements().get(0).getChildElements().get(1).getChildElements();
-		List<ServiceAttributes> group2Members = resourceElements.get(0).getChildElements().get(1).getChildElements().get(1).getChildElements();
-		List<ServiceAttributes> members = resourceElements.get(1).getChildElements();
-
-		//group has 3 members but member2 and member3 are expired
-		assertEquals(groupMembers.size(), 1);
-
-		//group2 has 2 members but member3 is expired
-		assertEquals(group2Members.size(), 1);
-
-		//resource has 3 members but member3 is expired in all groups of given resource
-		assertEquals(members.size(), 2);
-	}
-
```

Comment:
It's not strictly related to this pull-request, but I think we don't have much tests for the new methods. For example to test usage of expired members within the service. We might want to add them later.

Reply:
I think this might be discussed in the future and tasks to create these tests could be part of a sprint.